brew/brew.go

```diff
@@ -0,0 +1,150 @@
+package brew
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/manifoldco/promulgate/artifact"
+	"github.com/manifoldco/promulgate/git"
+)
+
+// NewBottles converts the provided zip file into a tarball suitable for use as
+// a brew bottle.
+func NewBottles(darwin *artifact.File, repo *git.Repository, tag string) ([]artifact.File, string, error) {
+	zr, err := zip.NewReader(darwin.ReaderAt(), darwin.Size)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if len(zr.File) != 1 {
+		return nil, "", errors.New("zip file must contain only 1 file")
+	}
+
+	var buf bytes.Buffer
+	gz := gzip.NewWriter(&buf)
+	defer gz.Close()
+	tgz := tar.NewWriter(gz)
+	defer tgz.Close()
+
+	fi := zr.File[0].FileInfo()
+	hdr, err := tar.FileInfoHeader(fi, "")
+	if err != nil {
+		return nil, "", err
+	}
+
+	parts := strings.SplitN(darwin.Name, "_", -1)
+
+	binname := hdr.Name
+	hdr.Name = filepath.Join(parts[0], parts[1], "bin", hdr.Name)
+	err = tgz.WriteHeader(hdr)
+	if err != nil {
+		return nil, "", err
+	}
+
+	zf, err := zr.File[0].Open()
+	if err != nil {
+		return nil, "", err
+	}
+
+	_, err = io.Copy(tgz, zf)
+	if err != nil {
+		return nil, "", err
+	}
+
+	var jbuf bytes.Buffer
+	err = receipt.Execute(&jbuf, receiptArgs{
+		Owner: repo.Owner,
+		Name:  repo.Name,
+		Tag:   parts[1],
+		Time:  time.Now().Unix(),
+	})
+	if err != nil {
+		return nil, "", err
+	}
+
+	hdr = &tar.Header{
+		Name: filepath.Join(parts[0], parts[1], "INSTALL_RECEIPT.json"),
+		Mode: 0600,
+		Size: int64(jbuf.Len()),
+	}
+	err = tgz.WriteHeader(hdr)
+	if err != nil {
+		return nil, "", err
+	}
+
+	_, err = io.Copy(tgz, &jbuf)
+	if err != nil {
+		return nil, "", err
+	}
+
+	tgz.Close()
+	gz.Close()
+
+	var bottles []artifact.File
+
+	for _, rel := range []string{"sierra", "el_capitan", "yosemite"} {
```

Comment:
Might be worth making this a constant `supportedOS` or something. That way in the godoc people can see what we support.